coffea/nanoevents/schemas/delphes.py

```diff
@@ -0,0 +1,296 @@
+            if {"fE", "fP"} == form.get("contents", {}).keys():
+                return zip_forms(
+                    {
+                        "x": form["contents"]["fP"]["contents"]["fX"],
+                        "y": form["contents"]["fP"]["contents"]["fY"],
+                        "z": form["contents"]["fP"]["contents"]["fZ"],
+                        "t": form["contents"]["fE"],
+                    },
+                    objname,
+                    "LorentzVector",
+                )
+            # If there's no "content", like a NumpyArray, just return.
+            # Note: this comes after checking for RecordArray.
+            if "content" not in form:
+                return form
+            # Then recursively go through and update the form's content.
+            form["content"] = _tlorentz_vectorize(objname, form["content"])
+            return form
+
+        # preprocess lorentz vectors properly (and recursively)
+        for objname, form in branch_forms.items():
+            branch_forms[objname] = _tlorentz_vectorize(objname, form)
+
+        # parse into high-level records (collections, list collections, and singletons)
+        collections = set(k.split("/")[0] for k in branch_forms)
+        collections -= set(k for k in collections if k.endswith("_size"))
+
+        # Create offsets virtual arrays
+        for name in collections:
+            if f"{name}_size" in branch_forms:
+                branch_forms[f"o{name}"] = transforms.counts2offsets_form(
+                    branch_forms[f"{name}_size"]
+                )
+
+        output = {}
+        for name in collections:
+            output[f"{name}.offsets"] = branch_forms[f"o{name}"]
+            mixin = self.mixins.get(name, "NanoCollection")
+
+            # Every delphes collection is a list
+            offsets = branch_forms["o" + name]
+            content = {
+                k[2 * len(name) + 2 :]: branch_forms[k]
+                for k in branch_forms
+                if k.startswith(name + "/" + name)
+            }
+            output[name] = zip_forms(content, name, record_name=mixin, offsets=offsets)
+
+            # update docstrings as needed
+            # NB: must be before flattening for easier logic
+            for parameter in output[name]["content"]["contents"].keys():
+                if "parameters" not in output[name]["content"]["contents"][parameter]:
+                    continue
+                output[name]["content"]["contents"][parameter]["parameters"][
+                    "__doc__"
+                ] = self.docstrings.get(
+                    parameter,
+                    output[name]["content"]["contents"][parameter]["parameters"].get(
+                        "__doc__", "no docstring available"
+                    ),
+                )
+
+            # handle branches named like [4] and [5]
+            output[name]["content"]["contents"] = {
+                k.replace("[", "_").replace("]", ""): v
```

Comment:
Since the extra dimension size can be inferred from the type, do you think it makes sense to drop also the number (i.e. `Thing` instead of `Thing_5` etc.) ? I'm ambivalent

Reply:
Initially I was thinking this, but I got worried that there might be a "PrunedP4" branch later and I just wanted to convert the non-python-dot-syntax-friendly characters to something more python-friendly. If this is something you're not necessarily concerned about, we could hard-code a regex replacement of `\[\d\]` instead.

Reply:
That's a good point, there may be a future name clash if for some reason `PrunedP4` and also `PrunedP4[5]` are available.

Reply:
`Tau[4]` also might name-clash for (obvious) reasons. So i was stuck here :-/

Reply:
What's the decision here? Shall we go ahead and merge as is?